geomstats/geometry/hyperbolic.py

```diff
@@ -3,38 +3,11 @@
 The n-dimensional hyperbolic space embedded and its different representations.
 """
 
-import math
-
 
 import geomstats.backend as gs
 from geomstats.geometry.embedded_manifold import EmbeddedManifold
-from geomstats.geometry.minkowski import Minkowski
-from geomstats.geometry.minkowski import MinkowskiMetric
 from geomstats.geometry.riemannian_metric import RiemannianMetric
 
-TOLERANCE = 1e-6
-
-SINH_TAYLOR_COEFFS = [0., 1.,
-                      0., 1 / math.factorial(3),
-                      0., 1 / math.factorial(5),
-                      0., 1 / math.factorial(7),
-                      0., 1 / math.factorial(9)]
-COSH_TAYLOR_COEFFS = [1., 0.,
-                      1 / math.factorial(2), 0.,
-                      1 / math.factorial(4), 0.,
-                      1 / math.factorial(6), 0.,
-                      1 / math.factorial(8), 0.]
-INV_SINH_TAYLOR_COEFFS = [0., - 1. / 6.,
-                          0., + 7. / 360.,
-                          0., - 31. / 15120.,
-                          0., + 127. / 604800.]
-INV_TANH_TAYLOR_COEFFS = [0., + 1. / 3.,
-                          0., - 1. / 45.,
-                          0., + 2. / 945.,
-                          0., -1. / 4725.]
-
-EPSILON = 1e-6
-
 
 class Hyperbolic(EmbeddedManifold):
```

Comment:
If we do not use the embedding for Hyperbolic, maybe make Hyperbolic inherit from Manifold directly? And delete the embedding_manifold parameter in the constructor?

Reply:
You point out the main problem. Here hyperbolic should be abstract but its two subclasses Hyperboloid and Poincaré,are quite differents if Hyperboloid needs to inherit from Embedding manifold (use Minkowsky), Poincaré Ball not, moreover both need to inherit methods from Hyperbolic.

Reply:
In this case, `Hyperboloid(Hyperbolic, EmbeddedManifold)` may work?

Reply:
while having `Hyperbolic(Manifold)` only?